src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -833,6 +835,26 @@ protected ForeignToRubyNode createForeignToRubyNode() {
 
     }
 
+    @CoreMethod(names = "to_java_array_inner", isModuleFunction = true, required = 1)
+    public abstract static class InteropToJavaArrayNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization(guards = "isRubyArray(array)")
+        public Object toJavaArray(DynamicObject array) {
+            return JavaInterop.asTruffleObject(ArrayStrategy.of(array).newMirror(array).copyArrayAndMirror().getArray());
```

Comment:
ArrayStrategy.of is neverPartOfCompilation() because it should always be cached on fast-path.
So either cache the strategy like in ArrayNodes or `@TruffleBoundary`.